src/telemetry.c

```diff
@@ -984,14 +976,6 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
 {
         size_t payload_len;
         int ret = 0;
-        int k = 0;
-        struct stat unused;
-
-        k = stat(TM_OPT_OUT_FILE, &unused);
-        if (k == 0) {
-                // Bail early if opt-out is enabled
-                return -ECONNREFUSED;
-        }
 
         payload_len = strlen((char *)payload);
 
```

Comment:
This is correct. Create record and set payload don't need to care about opt-in/opt-out status.